services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,19 +43,44 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  
+  if (event == NULL) {
+    return ERR_CODE_INVALID_ARG;
+  }
+  xQueueSend(thermalMgrQueueHandle, event, 0);
   return ERR_CODE_SUCCESS;
+  
 }
 
 void osHandlerLM75BD(void) {
+  
   /* Implement this function */
+  
+  float *temp;
+  readTempLM75BD(LM75BD_OBC_I2C_ADDR, temp);
+  if (*temp > 75.0) {
+    overTemperatureDetected();
+  }
+  else {
+    safeOperatingConditions();
+  } 
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  
+  float *temp;
   while (1) {
-    
+    thermal_mgr_event_t check; 
+    if (xQueueReceive(thermalMgrQueueHandle, pvParameters , 0) == pdTRUE) {
```

Comment:
There's no need to be polling to see if anything has entered the queue by setting the block time to 0. It would be better to have this block for longer and wait for something to be added to the queue  before continuing

Reply:
`pvParameters` isn't what you want to be passing in for the second argument of this function. The second parameter is where the received even from the queue will be stored, while `pvParameters` is a parameter that can optionally be passed into the task when it is created and then is never updated again